staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/timing_ratio_histogram.go

```diff
@@ -0,0 +1,272 @@
+	delegate := compbasemetrics.NewTestableTimingHistogramVec(nowFunc, opts, labelNames)
+	return &TimingRatioHistogramVec{
+		Registerable: delegate,
+		delegate:     delegate,
+	}
+}
+
+func (v *TimingRatioHistogramVec) metrics() Registerables {
+	return Registerables{v}
+}
+
+// NewForLabelValuesChecked will return an error if this vec is not hidden and not yet registered
+// or there is a syntactic problem with the labelValues.
+func (v *TimingRatioHistogramVec) NewForLabelValuesChecked(initialNumerator, initialDenominator float64, labelValues []string) (RatioedGauge, error) {
+	underMember, err := v.delegate.WithLabelValuesChecked(labelValues...)
+	if err != nil {
+		return noopRatioed{}, err
+	}
+	underMember.Set(initialNumerator / initialDenominator)
+	return &timingRatioHistogramInner{
+			delegate:    underMember,
+			numerator:   initialNumerator,
+			denominator: initialDenominator,
+		},
+		nil
+}
+
+// NewForLabelValuesSafe is the same as NewForLabelValuesChecked in cases where that does not
+// return an error.  When the unsafe version returns an error due to the vector not being
+// registered yet, the safe version returns an object that implements its methods
+// by looking up the relevant vector member in each call (thus getting a non-noop after registration).
+// In the other error cases the object returned here is a noop.
+func (v *TimingRatioHistogramVec) NewForLabelValuesSafe(initialNumerator, initialDenominator float64, labelValues []string) RatioedGauge {
+	tro, err := v.NewForLabelValuesChecked(initialNumerator, initialDenominator, labelValues)
+	if err == nil {
+		return tro
+	}
+	if !compbasemetrics.ErrIsNotRegistered(err) {
+		klog.ErrorS(err, "Failed to extract TimingRatioHistogramVec member, using noop instead", "vectorname", v.FQName(), "labelValues", labelValues)
+		return tro
+	}
+	return &timingRatioHistogramInner{
+		delegate:    &timingHistogramVecElt{delegate: v.delegate, labelValues: labelValues},
+		numerator:   initialNumerator,
+		denominator: initialDenominator,
+	}
+}
+
+// timingHistogramVecElt implements settable by looking up the relevant GaugeVecMetric in each Set call
+type timingHistogramVecElt struct {
+	delegate    compbasemetrics.GaugeVecMetric
+	labelValues []string
+}
+
+func (tve *timingHistogramVecElt) Set(x float64) {
+	tve.delegate.WithLabelValues(tve.labelValues...).Set(x)
+}
+
+type noopRatioed struct{}
+
+func (noopRatioed) Set(float64)            {}
+func (noopRatioed) Add(float64)            {}
+func (noopRatioed) Sub(float64)            {}
+func (noopRatioed) Inc()                   {}
+func (noopRatioed) Dec()                   {}
+func (noopRatioed) SetToCurrentTime()      {}
+func (noopRatioed) SetDenominator(float64) {}
+
+func (v *TimingRatioHistogramVec) Reset() {
+	v.delegate.Reset()
+}
+
+type TimingRatioHistogramPairVec struct {
+	urVec *TimingRatioHistogramVec
+}
```

Comment:
A more generic comment is - why do we even need this type?

The more types we have the harder it is to follow when to use what, I don't see a need for this type.

Why can't we just instead create a simple helper functions for the TiminingRatioHistogramVec type, like
```
// better name needed desired
func (v *TimingRatioHistogramVec) WaitinExecutingGaugePair(nominator, denominator labels) {
}
```

Reply:
It is a convenience.  See #110515 for its removal; not quite as small a thing as I suspect you were thinking.